src/core/lib/debug/trace.cc

```diff
@@ -104,15 +106,76 @@ void SavedTraceFlags::Restore() {
 }
 
 namespace {
+// A basic glob matcher based on https://research.swtch.com/glob.
+bool GlobMatch(absl::string_view name, absl::string_view trace_glob) {
```

Comment:
Let's move this to gprpp by itself (there are likely other things we can use this for)